pkg/controller/volume/persistentvolume/scheduler_binder.go

```diff
@@ -397,18 +464,26 @@ func (b *volumeBinder) checkBindings(pod *v1.Pod, bindings []*bindingInfo, claim
 		}
 
 		// Check if pvc is fully bound
-		if isBound, _, err := b.isPVCBound(binding.pvc.Namespace, binding.pvc.Name); !isBound || err != nil {
-			return false, err
+		if !b.isPVCFullyBound(pvc) {
+			return false, nil
 		}
-
-		// TODO; what if pvc is bound to the wrong pv? It means our assume cache should be reverted.
-		// Or will pv controller cleanup the pv.ClaimRef?
```

Comment:
https://github.com/kubernetes/kubernetes/blob/197fc67693c2391dcbc652fc185ba85b5ef82a8e/pkg/controller/volume/persistentvolume/pv_controller.go#L652-L699

If PVC is bound to wrong PV, PV will be unbound. Assumed cache will be reverted.
I've added a test for dynamic provisioned PV by the way.

Reply:
There is a PR here that will detect if PV has invalid node affinity and will fail binding: https://github.com/kubernetes/kubernetes/pull/70026